icu4c/source/i18n/units_converter.cpp

```diff
@@ -372,11 +372,20 @@ void U_I18N_API addSingleFactorConstant(StringPiece baseStr, int32_t power, Sign
         factor.constantExponents[CONSTANT_FT2M] += 3 * power * signum;
     } else if (baseStr == "in3_to_m3") {
         factor.constantExponents[CONSTANT_FT2M] += 3 * power * signum;
-        factor.factorDen *= 12 * 12 * 12;
+        if (signum == Signum::NEGATIVE) {
+            factor.factorNum *= 12 * 12 * 12;
+        } else {
+            factor.factorDen *= 12 * 12 * 12;
+        }
```

Comment:
Do we also need to consider `power` here?

Reply:
Excellent, yes.

We are doing it in Java but not here.

we should add a test case to test this case for the future.

Reply:
I see that Younies changed the implementation to consider power, but I don't see a new “test case to ensure this scenario is accounted for in the future.”

Reply:
This test need to be added to the CLDR: `unitsTest.txt` which should be in another PR: one in CLDR first and then one to ICU.

we need to add something like:

`square-gasoline-equivalent` to `square-kilograms-per-square-meter-p4-per-second`

Or something more simpler:

`square-gallon` to `p6-meter`.

or both.

Reply:
If you have an open CLDR ticket for adding that test case, then that should be good. I don't care if it's a ticket just for this, or an item in some other open ticket. Please reply with the CLDR ticket number.

We probably don't need a separate ICU ticket, assuming the next CLDR-to-ICU integration picks up the test data.

Reply:
sure, here is it

https://unicode-org.atlassian.net/browse/CLDR-17114